File: src/main/java/io/bindernews/thegrackle/Events.java
package io.bindernews.thegrackle;

import com.megacrit.cardcrawl.metrics.Metrics;
import com.megacrit.cardcrawl.screens.SingleCardViewPopup;
import io.bindernews.bnsts.eventbus.*;
import io.bindernews.thegrackle.api.IPopup;
import io.bindernews.thegrackle.api.SvcChangeCardEvent;

/**
 * A collection of {@link EventEmit}s, serving as a global event registry.
 */
public class Events {

    private Events() {}

    private static final EventEmit<SvcChangeCardEvent> svcCardChange = new EventEmit<>();
    private static final EventEmit<Metrics> metricsRun = new EventEmit<>();
    private static final HandlerList<IPopup> popups = new HandlerList<>();

    /** Event for when {@link SingleCardViewPopup} changes cards. */
    public static IEventEmit<SvcChangeCardEvent> svcCardChange() { return svcCardChange; }

    /** Event emitter, called at the end of {@link Metrics#run}. */
    public static IEventEmit<Metrics> metricsRun() { return metricsRun; }
    
    public static IHandlerList<IPopup> popups() { return popups; }
}

File: src/main/kotlin/io/bindernews/thegrackle/Events.kt
package io.bindernews.thegrackle

import com.megacrit.cardcrawl.metrics.Metrics
import com.megacrit.cardcrawl.screens.SingleCardViewPopup
import io.bindernews.bnsts.eventbus.EventEmit
import io.bindernews.bnsts.eventbus.HandlerList
import io.bindernews.bnsts.eventbus.IEventEmit
import io.bindernews.bnsts.eventbus.IHandlerList
import io.bindernews.thegrackle.api.IPopup
import io.bindernews.thegrackle.api.SvcChangeCardEvent

/**
 * A collection of [EventEmit]s, serving as a global event registry.
 */
object Events {

    /** Event for when [SingleCardViewPopup] changes cards.  */
    @JvmStatic val svcCardChange: IEventEmit<SvcChangeCardEvent> = EventEmit()
    /** Event emitter, called at the end of [Metrics.run].  */
    @JvmStatic val metricsRun: IEventEmit<Metrics> = EventEmit()
    /** List of UI popups */
    @JvmStatic val popups: IHandlerList<IPopup> = HandlerList()
}